test/e2e/csc_tests.go

```diff
@@ -0,0 +1,133 @@
+package e2e
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	operator "github.com/operator-framework/operator-marketplace/pkg/apis/operators/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/operator-framework/operator-sdk/pkg/test"
+)
+
+const (
+	testResourceName string = "test-operators"
+)
+
+func runCSCWithInvalidTargetNamespace(t *testing.T) {
+	ctx := test.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	// Get global framework variables
+	f := test.Global
+	// Run tests
+	if err := cscWithInvalidTargetNamespace(t, f, ctx); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func cscWithInvalidTargetNamespace(t *testing.T, f *test.Framework, ctx *test.TestCtx) error {
+	// Define variables to be used in test
+	targetNamespace := "nonexisting-namespace"
+	packages := "etcd,federationv2,mongodb,amq-streams,couchbase,descheduler,prometheus,svcat,clusterlogging,dynatrace,metering"
+
+	// Get test namespace
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("Could not get namespace: %v", err)
+	}
+
+	// Create the operatorsource to download the manifests
+	testOperatorSource := &operator.OperatorSource{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.OperatorSourceKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-operators",
+			Namespace: namespace,
+		},
+		Spec: operator.OperatorSourceSpec{
+			Type:              "appregistry",
+			Endpoint:          "https://quay.io/cnr",
+			RegistryNamespace: "marketplace_e2e",
```

Comment:
Given we are going to constantly add and remove this `OperatorSource`, please reduce the number of operators in the namespace.

Reply:
Rather than creating an operator source for each test, would it be better to use a single operator source for all tests that rely on it? I had spoken to Kevin about doing so in a future refactor PR: https://github.com/operator-framework/operator-marketplace/pull/145#discussion_r269083130

Reply:
+1

Reply:
Great - I will create a JIRA to refactor the code.